### PR TITLE
[14.0][MIG]&[FIX] remove_odoo_enterprise

### PR DESCRIPTION
**Pull request by clementmbr**
Third PR (based on #33) to migrate the module `remove_odoo_enterprise` but I hope this one is the right one :sparkles: 
cf #33 #38 

Basically there were 2 mistakes made in v13 :

- it's better to hide the field enterprise field rather than remove it because it can break some other part of the res.config.settings view (as it did for me when I installed `hr_timesheet`, which led me to make this PR)
- it's important to hide the enterprise field's grandparent and not great-grandparent in v14 otherwise it will hide other non-enterprise fields :

image

(to be honest I didn't check how is build the res.config.settings.view before v14, it can be interesting to check if it was not a problem in v13 too...)

...and I added some tests for code coverage.

cc @rvalyi

### CODE REVIEW

**Comment by rvalyi** [comment]
just to make things clear when the fields were removed brutally in the former PRs, you would have errors such as
`Error: While parsing modifiers for div: for modifier "invisible": Unknown field module_project_timesheet_synchro in domain`
because of view domains refering to these fields as when @clementmbr installed hr_timesheet

**Review by rvalyi — APPROVED** [review]

**Review by hparfr — APPROVED** [review]

File: remove_odoo_enterprise/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+{
+    "name": "Remove Odoo Enterprise",
+    "summary": "Remove enterprise modules and setting items",
+    "version": "14.0.1.0.0",
+    "category": "Maintenance",
+    "author": "Eska, Onestein, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/server-brand",
+    "license": "AGPL-3",
+    "depends": ["base"],
+    "external_dependencies": {"python": ["lxml"]},
```
Comment (astirpe) [inline remove_odoo_enterprise/__manifest__.py]: `lxml` it's already in standard Odoo: https://github.com/odoo/odoo/blob/14.0/requirements.txt#L22-L25
No need to add an external dependency in the manifest.

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# generated from manifests external_dependencies
```
Comment (astirpe) [inline requirements.txt:1]: This should no be created, as `lxml` is already required by standard odoo

**Review by astirpe — CHANGES_REQUESTED** [review]

File: remove_odoo_enterprise/__manifest__.py (hunk 1 above)
Comment (clementmbr) [reply remove_odoo_enterprise/__manifest__.py]: removed!

File: requirements.txt (hunk 2 above)
Comment (clementmbr) [reply requirements.txt:1]: Thanks!

**Pull request merged**